azure-pipelines.yml

```diff
@@ -42,18 +52,19 @@ stages:
     - checkout: sonic-mgmt-common
       clean: true
       submodules: recursive
-      displayName: 'Checkout code'
+      displayName: 'Checkout sonic-mgmt-common'
 
     - task: DownloadPipelineArtifact@2
       inputs:
         source: specific
         project: build
-        pipeline: 1
+        pipeline: 142
         artifact: sonic-buildimage.vs
         runVersion: 'latestFromBranch'
-        runBranch: 'refs/heads/master'
+        runBranch: 'refs/heads/$(BUILD_BRANCH)'
         patterns: |
           target/debs/buster/libyang*.deb
+          target/python-wheels/bullseye/sonic_yang_models*.whl
```

Comment:
Can you learn the Debian distribution on the fly instead?

Reply:
Do you have any references? I could not find another such instance.

Not sure how to determine it generically. Did not find any settings in `debian/control` or any other setting files under sonic-mgmt-framework repo. As per my understanding the target debian version is chosen buy the docker build rules file, by adding the docker image file name to different makefile variables -- like [here](https://github.com/sonic-net/sonic-buildimage/blob/master/rules/docker-sonic-mgmt-framework.mk#L41) and [here](https://github.com/sonic-net/sonic-buildimage/blob/master/rules/docker-gnmi.mk#L21C1-L21C23). These docker makefiles are not used during pipeline builds. In fact we do not download sonic-buildimage code as well. Even if we donwload we cannot invoke these makefiles directly. Also, the code pattern is not generic enough to grep & determine the debian version.